### PR TITLE
Map 1350 locations api create new prefix endpoint

### PR DESCRIPTION
**Pull request by GurnankCheema**
this will replace the call clients currently make to whereaboutApi's GET /locations/{agencyId}/{group}/location-prefix

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/resource/ApiExceptionHandler.kt [hunk 1]
```diff
@@ -313,6 +326,9 @@ class ApiExceptionHandler {
 }
 
 class LocationNotFoundException(id: String) : Exception("There is no location found for ID = $id")
+
+class LocationPrefixNotFoundException(id: String) : Exception("No mappings found for $id")
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/resource/ApiExceptionHandler.kt:330]: Think message could be clearer

**Review by Mjwillis — APPROVED** [review]
One minor comment.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/resource/ApiExceptionHandler.kt (hunk 1 above)
Comment (GurnankCheema) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/resource/ApiExceptionHandler.kt:330]: will change to "Location prefix not found for $id"

**Review by Mjwillis — APPROVED** [review]
LGTM

**Review by marcusaleman — COMMENTED** [review]
seen.

**Pull request merged**